smack-integration-test/src/main/java/org/jivesoftware/smackx/mood/MoodIntegrationTest.java

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.mood;
+
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.packet.Presence;
+import org.jivesoftware.smack.roster.PresenceEventListener;
+import org.jivesoftware.smack.roster.Roster;
+import org.jivesoftware.smackx.mood.element.MoodElement;
+
+import org.igniterealtime.smack.inttest.AbstractSmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.igniterealtime.smack.inttest.util.SimpleResultSyncPoint;
+import org.jxmpp.jid.BareJid;
+import org.jxmpp.jid.FullJid;
+import org.jxmpp.jid.Jid;
+
+public class MoodIntegrationTest extends AbstractSmackIntegrationTest {
+
+    public MoodIntegrationTest(SmackIntegrationTestEnvironment environment) {
+        super(environment);
+    }
+
+    @SmackIntegrationTest
+    public void test() throws Exception {
+        MoodManager mm1 = MoodManager.getInstanceFor(conOne);
+        MoodManager mm2 = MoodManager.getInstanceFor(conTwo);
+
+        Roster r1 = Roster.getInstanceFor(conOne);
+        Roster r2 = Roster.getInstanceFor(conTwo);
+
+        syncSubscribe(conOne, conTwo, "two");
+        syncSubscribe(conTwo, conOne, "one");
+
+        final SimpleResultSyncPoint moodReceived = new SimpleResultSyncPoint();
+
+        mm2.addMoodListener(new MoodListener() {
+            @Override
+            public void onMoodUpdated(BareJid jid, Message message, MoodElement moodElement) {
+                if (moodElement.getMood() == Mood.satisfied) {
+                    moodReceived.signal();
+                }
+            }
+        });
+
+        mm1.setMood(Mood.satisfied);
+
+        moodReceived.waitForResult(timeout);
+        r1.removeEntry(r1.getEntry(conTwo.getUser().asBareJid()));
```

Comment:
Those should be wrapped in a `finally` block, or even better, in a `@AfterClass` annotated method